pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -544,37 +547,72 @@ func (qjm *XController) GetAggregatedResourcesPerGenericItem(cqj *arbv1.AppWrapp
 	// Get all pods and related resources
 	for _, genericItem := range cqj.Spec.AggrResources.GenericItems {
 		itemsList, _ := genericresource.GetListOfPodResourcesFromOneGenericItem(&genericItem)
-		for i :=0; i < len(itemsList); i++ {
+		for i := 0; i < len(itemsList); i++ {
 			retVal = append(retVal, itemsList[i])
 		}
 	}
 
 	return retVal
 }
+
+func (qjm *XController) GetAllObjectsOwned(cqj *arbv1.AppWrapper) {
+
+	// Get all pods and related resources
+	countCompletedItems := 0
+	countCompletionRequired := 0
+	ignoredItems := 0
+	for _, genericItem := range cqj.Spec.AggrResources.GenericItems {
+		objectName := genericItem.GenericTemplate
```

Comment:
If the generic item does not have the new `completionstatus` field, can we skip this logic for performance reasons?  Please let me know your thoughts.

Reply:
Inside Appwrapper we are finding items of 3 types of items `completionRequired`, `completedItems` and `ignoredItems`, so we would need to iterate all the items I believe